clang/lib/Driver/ToolChains/ClassicFlang.cpp

```diff
@@ -82,12 +82,23 @@ void ClassicFlang::ConstructJob(Compilation &C, const JobAction &JA,
     Stem = C.getDriver().GetTemporaryPath("", "");
   } else {
     OutFile = Output.getFilename();
+    if (OutFile == "-.ll")
+      OutFile = "tmp";
     Stem = llvm::sys::path::filename(OutFile);
     llvm::sys::path::replace_extension(Stem, "");
   }
 
   // Add input file name to the compilation line
-  UpperCmdArgs.push_back(Input.getBaseInput());
+  const char *InputFile = Input.getBaseInput();
+  if (strcmp(InputFile, "-") == 0)
+  //FIXME:done specially for handling flang -E -dM - < /dev/null
```

Comment:
a small nit: all comments in this file have one space after `//`, e.g. `// FIXME:...`

Reply:
@pawosm-arm fixed. Thanks